cognee/infrastructure/databases/graph/neptune_driver/adapter.py

```
@@ -0,0 +1,1445 @@
+                    **neighbor["properties"]
+                }
+                for neighbor in result
+            ]
+
+            logger.debug(f"Retrieved {len(neighbors)} neighbors for node: {node_id}")
+            return neighbors
+
+        except Exception as e:
+            error_msg = format_neptune_error(e)
+            logger.error(f"Failed to get neighbors for node {node_id}: {error_msg}")
+            raise Exception(f"Failed to get neighbors: {error_msg}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)